kotlinx-coroutines-core/common/src/Delay.kt

```diff
@@ -75,5 +77,31 @@ public suspend fun delay(timeMillis: Long) {
     }
 }
 
+/**
+ * Delays coroutine for a given [duration] without blocking a thread and resumes it after the specified time.
+ * This suspending function is cancellable.
+ * If the [Job] of the current coroutine is cancelled or completed while this suspending function is waiting, this function
+ * immediately resumes with [CancellationException].
+ *
+ * Note that delay can be used in [select] invocation with [onTimeout][SelectBuilder.onTimeout] clause.
+ *
+ * Implementation note: how exactly time is tracked is an implementation detail of [CoroutineDispatcher] in the context.
+ */
+@ExperimentalTime
+public suspend fun delay(duration: Duration) = delay(duration.toDelayMillis())
+
 /** Returns [Delay] implementation of the given context */
 internal val CoroutineContext.delay: Delay get() = get(ContinuationInterceptor) as? Delay ?: DefaultDelay
+
+/**
+ * Convert this duration to its millisecond value.
+ * Positive durations are coerced at least `1`.
+ */
+@ExperimentalTime
+internal fun Duration.toDelayMillis(): Long =
+        when {
+            this > Duration.ZERO -> toLongMilliseconds().coerceAtLeast(1)
+            this < Duration.ZERO -> -1
```

Comment:
I've reviewed usages of `toDelayMillis()` and could not find a case which cares about the distinction between zero and a negative value (either both mean the same or both are not supported), so this branch of `when` seems redundant (negative values can be also mapped to zero).

Reply:
Ok, test passes